---
layout: page
title: Assembly of metagenomic sequencing data
summary: "Why metagenomic data should be assembled? What is the difference between co-assembly and individual assembly?
What is the difference between reads, contigs and scaffolds?
How to assess the quality of metagenomic data assembly? 
"
---

---

Metagenomics involves the extraction, sequencing and analysis of combined genomic DNA from entire microbiome samples. 
It includes then DNA from **many different organisms**, with different taxonomic background.

Reconstructing the genomes of microorganisms in the sampled communities is critical step in analyzing metagenomic data. 
To do that, we can use **assembly** and assemblers, 
i.e. computational programs that stich together the small fragments of sequenced DNA produced by sequencing instruments.

Assembling seems intuitively similar to putting together a jigsaw puzzle. 

**Metagenomic assembly** is further complicated by:

* the large volume of data produced 
* the quality of the sequence 
* the unequal representation of members of the microbial community 
* the presence of closely related microorganisms with similar genomes 
* the presence of several strains of the same microorganism 
* an insufficient amount of data for minor community members

For assembly, there are 3 main strategies:

1. Greedy extension 
2. Overlap Layout Consensus 
3. De Bruijn graphs. 

![]({{site.url}}/images/assembly-algorithms.png)

In this tutorial, 
we will learn how to run metagenomic assembly tool and evaluate the quality of the generated assemblies. 

We will use data from the study: 
[Temporal shotgun metagenomic dissection of the coffee fermentation ecosystem](https://www.ebi.ac.uk/metagenomics/studies/MGYS00005630#overview). 

For an in-depth analysis of the structure and functions of the coffee microbiome, 
a temporal shotgun metagenomic study (six time points) was performed. 
The six samples have been sequenced with Illumina MiSeq utilizing whole genome sequencing.


<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Preparing data

1. Create a new history and rename it
2. Import data
```shell
   https://zenodo.org/record/7818827/files/ERR2231567_1.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231567_2.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231568_1.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231568_2.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231569_1.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231569_2.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231570_1.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231570_2.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231571_1.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231571_2.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231572_1.fastqsanger.gz
   https://zenodo.org/record/7818827/files/ERR2231572_2.fastqsanger.gz 
```
3. Create a paired collection named Raw reads, rename your pairs with the sample name
   - Click on **Select Items** at the top of the history panel
   ![]({{site.url}}/images/historyItemControls.png)
   - Check all the datasets in your history you would like to include 
   - Click **n of N selected** and choose **Build List of Dataset Pairs**
   - Change the text of *unpaired forward* to a common selector for the forward reads
   - Change the text of *unpaired reverse* to a common selector for the reverse reads 
   - Click **Pair these datasets** for each valid forward and reverse pair. 
   - Enter a name for your collection 
   - Click **Create List** to build your collection 
   - Click on the checkmark icon at the top of your history again

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Assembly

As explained before, there are many challenges to metagenomics assembly, including:

1. differences in coverage between samples, resulting from differences in abundance, 
2. the fact that different species often share conserved regions 
3. the presence of multiple strains of a single species

To reduce the differences in coverage between samples, 
we can use a **co-assembly** approach, where reads from all samples are aligned together:

![]({{site.url}}/images/co-assembly.png)

Pros of co-assembly | Cons of co-assembly
------|-------------------
More data | Higher computational overhead
Better/longer assemblies | Risk of shattering your assembly
Access to lower abundant organisms | Risk of increased contamination

**Co-assembly** is reasonable if:

* Same samples
* Same sampling event 
* Longitudinal sampling of the same site 
* Related samples

If it is not the case, **individual assembly** should be preferred. 
In this case, an extra step of **de-replication** should be used

Co-assembly is more commonly used than individual assembly and then de-replication after binning. 
But in this tutorial, to show all steps, we will run an **individual assembly**.

Several tools are available for metagenomic assembly. But 2 are the most used ones:

* [**MetaSPAdes**](https://doi.org/10.1101/gr.213959.116): an short-read assembler designed specifically for large and complex metagenomics datasets 
* [**MEGAHIT**](https://doi.org/10.1093/bioinformatics/btv033): a single node assembler for large and complex metagenomics NGS reads, such as soil

Both tools are available in Galaxy. But currently, only `MEGAHIT` can be used in individual mode for several samples.

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

### Individual assembly of short-reads with MEGAHIT
1. `MEGAHIT` Tool with parameters
   - *Select your input option*: Paired-end collection 
   - *Run in batch mode?*: Run individually
   - *Select a paired collection: Raw reads
   - *Basic assembly options*
     - *K-mer specification method*: Specify min, max, and step values 
     - *Minimum kmer size*: 21 
     - *Maximum kmer size*: 91 
     - *Increment of kmer size of each iteration*: 12
2. Don't RUN
3. Since the assembly process would take **~1h** we are just going to import the results of the assembly previously run.
```shell
https://zenodo.org/record/7818827/files/contigs_ERR2231567.fasta
https://zenodo.org/record/7818827/files/contigs_ERR2231568.fasta
https://zenodo.org/record/7818827/files/contigs_ERR2231569.fasta
https://zenodo.org/record/7818827/files/contigs_ERR2231570.fasta
https://zenodo.org/record/7818827/files/contigs_ERR2231571.fasta
https://zenodo.org/record/7818827/files/contigs_ERR2231572.fasta
```
4. Create a collection named **MEGAHIT Contig**, rename your pairs with the sample name

`MEGAHIT` produced a collection of output assemblies - one per sample - 
that can be proceeded further in binning step and then de-replication. 

The output contains **contigs**, contiguous lengths of genomic sequences in which bases are known to a high degree of certainty.

<span class="fa fa-fw fa-question-circle"></span> **Question**
1. How many contigs has been for ERR2231568 sample? 
2. And for ERR2231572? 
3. What is the minimum length of the contigs?

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

### Co-assembly with MetaSPAdes

1. `MetaSPAdes` Tool with following parameters
   - *Pair-end reads input format*: Paired-end: list of dataset pairs 
     - *FASTQ file(s): collection*: Raw reads 
   - *Select k-mer detection option*: User specific 
     - *K-mer size values*: 21,33,55,77

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Quality control of assembly


1. `Quast` Tool with parameters
    - *Assembly mode?*: Individual assembly (1 contig file per samples)
      - *Use customized names for the input files?*: No, use dataset names 
        - *Contigs/scaffolds file*: output `MEGAHIT`
      - *Reads options*: Illumina paired-end reads in paired collection (To make the job quicker, you can select Disabled here. The raw reads will then not been mapped to the assembly to compute metrics, like the coverage.)
        - *FASTQ/FASTA files*: Raw reads 
    - *Type of assembly*: Metagenome
    - *Output files*: HTML report, PDF report, Tabular reports, Log file, Key metric summary (metagenome mode), Krona charts (metagenome mode without reference genomes)
2. Inspect the HTML reports

<img src="{{site.url}}/images/quast.png">

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

### Assembly statistics
1. **Genome statistics**
    - **Genome fraction (%):** percentage of aligned bases in the reference genome
    - **Duplication ratio:** total number of aligned bases / genome fraction * reference length
2. **Read mapping**: results of the mapping of the raw reads on the different assemblies (only if the *Reads options* is not disabled)
3. **Misassemblies**: joining sequences that should not be adjacent.
4. **Mismatches** or mismatched bases in the contig-reference alignment
5. Statistics without reference
    - **# contigs:** total number of contigs
    - **Largest contig:** length of the longest contig in the assembly
    - **N50:** length for which the collection of all contigs of that length or longer covers at least half an assembly
    - **L50:** number of contigs equal to or longer than N50

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

### Icarus contig browser

<img src="{{site.url}}/images/contig-size-viewer.png">

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Conclusion
Metagenomic data can be assembled to, ideally, obtain the genomes of the species that are represented within the input data. But metagenomic assembly is complex and there are

* **different approaches** like de Bruijn graphs methods
* **different strategies**, such as co-assembly, when we assembly all samples together, and individual assembly, when we assembly samples one by one
* **different** tools like `MetaSPAdes` and `MEGAHIT`













