---
layout: page
title: Hands-On
summary: "Where you'll apply the knowledge and techniques you've acquired throughout the previous modules. 
"
---

---

In this engaging "Hands-On" section, 
we embark on an exciting journey to replicate the genomic analysis performed in the study titled 
[Genomic Surveillance Reveals Antibiotic Resistance Gene Transmission via Phage Recombinases 
within Sheep Mastitis-Associated Streptococcus uberis](https://www.researchsquare.com/article/rs-1560496/v1). 

This study, conducted on 46 S. uberis isolates collected from mastitis-infected sheep in Sardinia, Italy, 
serves as our real-world case study.

* Genomes were assigned into lineage clusters using `PopPUNK`, which found **27 distinct isolate
clusters**, indicating considerable genetic variability consistent with environmental isolates. Geographic
trends were identified including regional linkage of several isolate clusters. Multi-locus Sequence Typing
(`MLST`) performed poorly and provided no new insights.
* Genomes were then screened for **antimicrobial resistance genes**, which were compared to phenotypic resistance profiles. Isolates showed **consistent phenotypic resistance to aminoglycosides** with variable **resistance to novobiocin and tetracycline**. A  multi-antimicrobial resistance cassette (**aminoglycoside, lincosamide and streptogramin**) was identified in the chromosome of **three genomes**.
* Genomes were screened for putative **virulence factors**, which identified **16 genes conserved between sheep and cow isolates**, with no host-specific genes shared uniformly across all host-specific isolates.
* **Pangenomic analysis** was then performed to identify core genes which were putatively surface-exposed, for identification of potential vaccine targets. As all genomes encoded sortase, core genes were screened for the sortase cleavage motif. Of the 1445 core S. uberis genes, 64 were putative sortase substrates and were predominantly adhesins, permeases and peptidases, consistent with compounds found within ruminant milk such as xanthine, fibronectin and lactoferrin. 


<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Get Data

Retrieve the **BioProject ID** from the article and  upload the datasets to a new history in Galaxy

In order to save time, we have pre-selected the samples that appeared to be the most significant. 

```text
SRR18739533 
SRR18739534
SRR18739542
SRR18739550
SRR18739557
SRR18739561
SRR18739563
SRR18739566
SRR18739571
```
<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Quality Check

1. **FastQC** & **MultiQC**
    <img src="{{site.url}}/images/hd-qc.png" width="100%">
3. **Cutadapt** or **TrimGalore!** with **q>=25**
1. **FastQC** & **MultiQC**

<img src="{{site.url}}/images/hd-trim.png" width="100%">

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Taxonomic Profiling

1. **Kraken2**
2. **Bracken**
3. **KrakenTools**
4. **Krona Pie Chart**
<img src="{{site.url}}/images/hd-kraken-krona.png" width="100%">
5. **Kraken-Biom** & **Phinch**
<img src="{{site.url}}/images/hd-kraken-phinch.png" width="100%">
6. **Pavian**
<img src="{{site.url}}/images/hd-kraken-pavian.png" width="100%">
<img src="{{site.url}}/images/hd-kraken-pavian-1.png" width="100%">
7. **MetaPhlAn**
8. **Krona Pie Charts**
<img src="{{site.url}}/images/hd-metaphlan-krona.png" width="100%">
9. **GraPhlAn**
<img src="{{site.url}}/images/hd-metaphlan-graphlan.png" width="45%">

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---
## Assembly

**NOTE!!!** 
* `metaSPAdes` only accepts paired-end reads.  
* `MEGAHIT`  accepts single reads but for single reads can be used in individual mode for several samples.
* `SPAdes` provides *Ion Torrent*  features
* `Unicycler` can be used in individual mode for several samples, even if it is not expressly developed for ion torrent reads

- `SPAdes`
  1. **SPADes** (one-o-one) with following parameters:
    - *k-mer*: 27,53,71,87,87,99,111,119,127
    - --isolate
    - --iontorrent
  2. **QUAST** and **Bowtie2**
  3. **MultiQC**

<img src="{{site.url}}/images/hd-spades-quast_num_contigs.png" width="100%">
<img src="{{site.url}}/images/hd-spades-bowtie2_se_plot.png" width="100%">

- `Unicycler` 
  1. **Unicycler** with the following parameters:
     - *k-mer*: 27,53,71,87,99,111,119,127 
     - only keep final files
  2. **QUAST** and **Bowtie2**
  3. **MultiQC**


<img src="{{site.url}}/images/hd-quast-1.png" width="100%">
<img src="{{site.url}}/images/hd-quast-2.png" width="100%">
<img src="{{site.url}}/images/hd-quast-3.png" width="100%">

<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---

## Pangenome Analysis

1. **Roary**
2. **Phyogenetic reconstruction with RAxML**
3. **Phandango**

<img src="{{site.url}}/images/hd-roary.png" width="100%">
<img src="{{site.url}}/images/hd-phandango-1.png" width="100%">
<img src="{{site.url}}/images/hd-phandango-2.png" width="100%">


<p style="text-align:right"><a href="{{site.url}}{{page.url}}"><strong>Go Up</strong><span class="fa fa-fw fa-arrow-up"></span></a></p>
---
